homeassistant/components/lg_soundbar/__init__.py

```diff
@@ -1 +1,36 @@
 """The lg_soundbar component."""
+import logging
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .config_flow import test_connect
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Set up platform from a ConfigEntry."""
+    hass.data.setdefault(DOMAIN, {})
+    # Verify the device is reachable before setting up the platform
+    try:
+        test_connect(entry.options[CONF_HOST], entry.options[CONF_PORT])
```

Comment:
This worked for me. Never touched any of this before, but I compared this integration to other integrations, and saw that others used `entry.data`.
```suggestion
        test_connect(entry.data[CONF_HOST], entry.data[CONF_PORT])
```

Reply:
thanks im also new to the code base and have not wroked with python apart from back in grade school. 
looks like options has the data when you 1st submit the config_flow then it is in data. i also added the condition to confirm that we have a host and port in entry.data before we test the connection